x-pack/plugins/reporting/server/lib/screenshots/wait_for_visualizations.ts

```diff
@@ -11,10 +11,23 @@ import { HeadlessChromiumDriver } from '../../browsers';
 import { LayoutInstance } from '../layouts';
 import { CONTEXT_WAITFORELEMENTSTOBEINDOM } from './constants';
 
-type SelectorArgs = Record<string, string>;
+interface CompletedItemsCountParameters {
+  context: string;
+  count: number;
+  renderCompleteSelector: string;
+}
 
-const getCompletedItemsCount = ({ renderCompleteSelector }: SelectorArgs) => {
-  return document.querySelectorAll(renderCompleteSelector).length;
+const getCompletedItemsCount = ({
+  context,
+  count,
+  renderCompleteSelector,
+}: CompletedItemsCountParameters) => {
+  const { length } = document.querySelectorAll(renderCompleteSelector);
```

Comment:
I was guessing the root cause of https://github.com/elastic/kibana/issues/117425 was that the query selector we pass in could be a bit too non-specific, so performance suffers more and more as the DOM is larger. That could still be valid, but I didn't want to block this PR.

Filed: https://github.com/elastic/kibana/issues/118098